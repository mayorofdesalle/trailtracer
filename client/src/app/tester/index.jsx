import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
