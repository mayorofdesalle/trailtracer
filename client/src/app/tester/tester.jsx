import { render } from '@testing-library/react'
import { Provider } from 'react-redux';

import './mocks/matchMedia.mock';
import store from '@app/store';
import Theme from '@components/style/Theme'

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                {children}
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };