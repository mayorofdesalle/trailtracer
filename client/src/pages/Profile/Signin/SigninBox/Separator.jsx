import { useContext } from 'react';
import styled from 'styled-components';

import Container from '@components/ui/Container';
import Text from '@components/ui/Text';
import PageContext from '@context/PageContext';

const HR = styled.hr`
    width: 100%;
    height: 2px;
    margin-left: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const SeparatorInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${HR}:first-of-type {
        margin-left: 0;
        margin-right: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    }

    & > ${Text} {
        flex-shrink: 0;
    }
`;

/**
 * Separator
 * @description
 * Horizontal rule with text in the middle.
 * **/
const Separator = () => {
    const context = useContext(PageContext);
    const theme = context.theme;
    const t = context.t;

    return (
        <SeparatorInner>
            <HR /><Text $color={theme.colors.background}>{t('signinPage.auths')}</Text><HR />
        </SeparatorInner>
    );
};

export default Separator;