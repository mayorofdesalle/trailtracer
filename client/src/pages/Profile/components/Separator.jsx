import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

import { Container } from '@components/ui';
import { Text } from '@components/ui';

const HR = styled.hr`
    width: 100%;
    height: 2px;
    margin-left: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const SeparatorInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${HR}:first-of-type {
        margin-left: 0;
        margin-right: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    }

    & > ${Text} {
        flex-shrink: 0;
    }
`;

/**
 * Separator
 * @description
 * Horizontal rule with text in the middle.
 * **/
const Separator = ({ color, ...props }) => {
    const { t } = useTranslation();

    return (
        <SeparatorInner {...props} role='separator'>
            <HR role='none' /><Text $color={color}>{t('signinPage.auths')}</Text><HR role='none' />
        </SeparatorInner>
    );
};

Separator.propTypes = {
    color: PropTypes.string
};

export default Separator;