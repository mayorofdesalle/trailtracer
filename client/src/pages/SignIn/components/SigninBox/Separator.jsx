import styled, { useTheme } from 'styled-components';

import Container from '@components/ui/Container';
import Text from '@components/ui/Text';

const HR = styled.hr`
    width: calc((100% - clamp(0.825rem, min(2dvw, 2dvh), 2.5625rem) * 9) / 2);
    height: 2px;
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const SeparatorInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${HR} {
        margin-left: 0.5rem;
    }

    & > ${HR}:first-of-type {
        margin-left: 0;
        margin-right: 0.5rem;
    }
`;

const Separator = () => {
    const theme = useTheme();
    return (
        <SeparatorInner>
            <HR /><Text $color={theme.colors.background}>or continue with</Text><HR />
        </SeparatorInner>
    );
};

export default Separator;