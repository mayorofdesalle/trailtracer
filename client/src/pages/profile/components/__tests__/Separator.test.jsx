import { describe, expect, it, render, screen } from '@app/tester';

import Separator from '../Separator';

describe('Separator', () => {

    it('should render correctly', () => {
        render(<Separator />);
        const separator = screen.getByRole('separator');
        expect(separator).toBeInTheDocument();
    });

});