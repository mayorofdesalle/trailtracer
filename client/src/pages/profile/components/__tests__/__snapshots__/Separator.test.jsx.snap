// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-irLvIq kHwwsn jDqJij"
    role="separator"
  >
    <hr
      class="sc-kFCroH bIVWxK"
      role="none"
    />
    <div
      class="sc-gLLuof fyoZns"
    >
      signinPage.auths
    </div>
    <hr
      class="sc-kFCroH bIVWxK"
      role="none"
    />
  </div>
</DocumentFragment>
`;
